internal/controllers/machineset/machineset_controller_status_test.go

```diff
@@ -0,0 +1,625 @@
+			machineSet: machineSet,
+			machines:   nil,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.MachineSetMachinesReadyInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:       "no machines",
+			machineSet: machineSet,
+			machines:   []*clusterv1.Machine{},
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status: metav1.ConditionTrue,
+				Reason: clusterv1.MachineSetMachinesReadyNoReplicasV1Beta2Reason,
+			},
+		},
+		{
+			name:       "all machines are ready",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("machine-1", readyCondition),
+				createMachine("machine-2", readyCondition),
+			},
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status: metav1.ConditionTrue,
+				Reason: v1beta2conditions.MultipleInfoReportedReason,
+			},
+		},
+		{
+			name:       "one ready, one has nothing reported",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("machine-1", readyCondition),
+				createMachine("machine-2"),
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  v1beta2conditions.NotYetReportedReason,
+				Message: "Condition Ready not yet reported from Machine machine-2",
+			},
+		},
+		{
+			name:       "one ready, one reporting not ready, one reporting unknown, one reporting deleting",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("machine-1", readyCondition),
+				createMachine("machine-2", metav1.Condition{
+					Type:    clusterv1.MachineReadyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  "SomeReason",
+					Message: "HealthCheckSucceeded: Some message",
+				}),
+				createMachine("machine-3", metav1.Condition{
+					Type:    clusterv1.MachineReadyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  "SomeUnknownReason",
+					Message: "Some unknown message",
+				}),
+				createMachine("machine-4", metav1.Condition{
+					Type:    clusterv1.MachineReadyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  clusterv1.MachineDeletingV1Beta2Reason,
+					Message: "Deleting: Machine deletion in progress, stage: DrainingNode",
+				}),
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesReadyV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
+				Reason:  v1beta2conditions.MultipleIssuesReportedReason,
+				Message: "Deleting: Machine deletion in progress, stage: DrainingNode from Machine machine-4; HealthCheckSucceeded: Some message from Machine machine-2; Some unknown message from Machine machine-3",
```

Comment:
@fabriziopandini Not for this PR, but in general this is hard to parse. A listing like we have in Node drain would make this a lot easier